### PR TITLE
Platform Python Diff

### PR DESCRIPTION
**Pull request by torsava**
DO NOT MERGE.

### CODE REVIEW

File: 00102-lib64.patch [hunk 1]
```diff
@@ -65,30 +65,30 @@ index b9bbfe5..2a5f29c 100644
...
+-        'stdlib': '{installed_base}/lib/platform-python{py_version_short}',
+-        'platstdlib': '{platbase}/lib/platform-python{py_version_short}',
++        'stdlib': '{installed_base}/lib64/platform-python{py_version_short}',
++        'platstdlib': '{platbase}/lib64/platform-python{py_version_short}',
+         'purelib': '{base}/lib/platform-python{py_version_short}/site-packages',
+-        'platlib': '{platbase}/lib/platform-python{py_version_short}/site-packages',
++        'platlib': '{platbase}/lib64/platform-python{py_version_short}/site-packages',
+         'include': '{installed_base}/include/platform-python{py_version_short}{abiflags}',
+         'scripts': '{base}/bin',
+         'data': '{base}',
```
Comment (torsava) [inline 00102-lib64.patch:91]: @encukou Could I ask for a short summary why `{userbase}` was changed to several different locations?

File: platform-python.spec [hunk 2]
```diff
@@ -471,22 +421,11 @@ BuildRoot: %{_tmppath}/%{name}-%{version}-root
 URL: https://www.python.org/
 
 # See notes in bug 532118:
-Provides: python(abi) = %{pybasever}
+Provides: platform-python(abi) = %{pybasever}
 
-Requires: %{name}-libs%{?_isa} = %{version}-%{release}
+Provides: bundled(python3-setuptools) = 28.8.0
```
Comment (hroncok) [inline platform-python.spec]: Remove bundled setuptools?

File: platform-python.spec [hunk 3]
```diff
@@ -994,10 +831,24 @@ sed -i -e "s/'pyconfig.h'/'%{_pyconfig_h}'/" \
...
+
+# Create symlinks of the platform-python binary
+
+ln -s \
+  %{_libexecdir}/platform-python \
+  %{buildroot}%{_libexecdir}/platform-python3
+
+ln -s \
+  %{_libexecdir}/platform-python3 \
+  %{buildroot}%{_libexecdir}/platform-python%{pybasever}
```
Comment (hroncok) [inline platform-python.spec]: I think the symblinks should go in the other direction.

File: platform-python.spec [hunk 4]
```diff
@@ -1226,109 +1017,27 @@ rm -fr %{buildroot}
...
 %license LICENSE
 %doc README.rst
-%{_bindir}/pydoc*
-%{_bindir}/python3
-%{_bindir}/python%{pybasever}
-%{_bindir}/python%{pybasever}m
-%{_bindir}/pyvenv
-%{_bindir}/pyvenv-%{pybasever}
-%{_mandir}/*/*
+%exclude %{_bindir}/pydoc*
```
Comment (hroncok) [inline platform-python.spec]: Can we exclude the entire bindir with one line?

File: 00102-lib64.patch (hunk 1 above)
Comment (encukou) [reply 00102-lib64.patch:91]: `posix_user` is config for installing Python modules into a user's home (`~/.local`). That's not supported for platform-python, so `posix_user` was changed to match `posix_prefix`.

**Comment by encukou** [comment]
In my mock build:
- Nothing requires /usr/bin/python3.6
- Nothing provides python(abi)

File: platform-python.spec [hunk 5]
```diff
@@ -472,131 +388,83 @@ BuildRoot: %{_tmppath}/%{name}-%{version}-root
...
-- the libpython dynamic library, for use by applications that embed Python as
-a scripting language, and by the main "python3" executable
-- the Python standard library
 
-%package -n system-python
-Summary:        System Python executable
-Group:          Development/Libraries
-Requires:       system-python-libs%{?_isa} = %{version}-%{release}
-Provides:       system-python(abi) = %{pybasever}
+%define __requires_exclude ^(/usr/bin/python3.*|python\\(abi\\) = 3\\..*)$
```
Comment (hroncok) [inline platform-python.spec]: Is this really needed?

If anything has `/usr/bin/python3.6` shebang, we should sed it out. Also, nothing should provide `python(abi) = 3.6` by any mean and if it does, something is wrong with the scripts responsible (in that case, we should probably keep that half for now, but add a TODO comment).

File: platform-python.spec [hunk 6]
```diff
@@ -1227,109 +950,22 @@ rm -fr %{buildroot}
 
 %postun libs -p /sbin/ldconfig
 
-%post -n system-python-libs -p /sbin/ldconfig
-
-%postun -n system-python-libs -p /sbin/ldconfig
+%post libs-devel -p /sbin/ldconfig
```
Comment (hroncok) [inline platform-python.spec]: Not needed.

File: platform-python.spec [hunk 7]
```diff
@@ -1227,109 +950,22 @@ rm -fr %{buildroot}
...
-
-%postun
-if [ $1 -eq 0 ] ; then
-    /bin/touch --no-create %{_datadir}/icons/hicolor &>/dev/null
-    /usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
-fi
-
-%posttrans
-/usr/bin/gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
+%postun libs-devel -p /sbin/ldconfig
```
Comment (hroncok) [inline platform-python.spec]: Not needed.

File: platform-python.spec [hunk 8]
```diff
@@ -1470,57 +1106,101 @@ fi
...
-%if 0%{?with_systemtap}
-%dir %(dirname %{tapsetdir})
-%dir %{tapsetdir}
-%{tapsetdir}/%{libpython_stp_optimized}
-%doc systemtap-example.stp pyfuntop.stp
-%endif
+%{_libdir}/libplatform-python3.so
+
+%files libs-devel
+%defattr(-,root,root,-)
```
Comment (hroncok) [inline platform-python.spec]: Is this needed? I think it's EL6 or even EL5 thing, and is default for now, but maybe I'm wrong.

File: platform-python.spec [hunk 9]
```diff
@@ -1470,57 +1106,101 @@ fi
...
+%{_rpmconfigdir}/macros.d/macros.pybytecompile-%{name}
 
 %files tools
 %defattr(-,root,root,755)
-%{_bindir}/python3-2to3
-%{_bindir}/2to3-%{pybasever}
-%{_bindir}/idle*
+%exclude %{_bindir}/python3-2to3
+%exclude %{_bindir}/2to3-%{pybasever}
+%exclude %{_bindir}/idle*
```
Comment (hroncok) [inline platform-python.spec]: Proposal: Use a oneliner here or nothing at all (already excluded somewhere else).

File: platform-python.spec (hunk 8 above)
Comment (torsava) [reply platform-python.spec]: It's in all the %files sections in python3. I agree we should remove it, but I'd say first in python3.

File: platform-python.spec (hunk 5 above)
Comment (torsava) [reply platform-python.spec]: While I agree about both points, I think it's safer to just leave it as is as a safety measure.

File: platform-python.spec [hunk 10]
```diff
@@ -2,46 +2,12 @@
...
-# Then the most important packages have to be built, starting from their
-# various leaf dependencies recursively. After these have been built, a
-# targeted rebuild should be requested for the rest.
-#
-# Currently these packages are recommended to have been built before a targeted
-# rebuild after a python abi change:
-#   python-sphinx, pytest, python-requests, cloud-init, dnf, anaconda, abrt
-
-%global with_rewheel 1
-
```
Comment (torsava) [inline platform-python.spec:38]: It would be great to link to the Platform Python Stack change here on top, for quick reference.

Comment (encukou) [reply platform-python.spec:38]: I don't think the Change will be relevant enough in the future -- it's full of rationale on bringing this package to Fedora.

Do you think the `%description` doesn't say enough?
```
Python interpreter for low-level system tools, designed to work with
the Platform module.
```

File: platform-python.spec (hunk 8 above)
Comment (encukou) [reply platform-python.spec]: Removed; added removal in python3 to my TODO

File: platform-python.spec (hunk 5 above)
Comment (encukou) [reply platform-python.spec]: Removed together with a forgotten bad shebang

File: platform-python.spec [hunk 11]
```diff
@@ -941,17 +941,9 @@ CheckPython optimized
 
 
 # ======================================================
-# Scriptlets
+# Files
 # ======================================================
 
-%post libs -p /sbin/ldconfig
-
-%postun libs -p /sbin/ldconfig
```
Comment (torsava) [inline platform-python.spec]: @encukou Why did you remove this as well? I believe it's useless only for libs-devel, for libs it serves a function.

Comment (encukou) [reply platform-python.spec]: Oops! Good catch

**Comment by torsava** [comment]
It took a while to get through it all, but I believe it's good to go!

File: platform-python.spec [hunk 12]
```diff
@@ -405,8 +405,6 @@ Summary:        Platform Python runtime libraries
 # yet upgraded expat:
 Requires: expat >= 2.1.0
 
-%define __requires_exclude ^(/usr/bin/python3.*|python\\(abi\\) = 3\\..*)$
```
Comment (hroncok) [inline platform-python.spec]: Have you checked final requires?

Comment (encukou) [reply platform-python.spec]: yes

**Comment by hroncok** [comment]
@torsava Can I close this and archive this repo?

**Comment by torsava** [comment]
> @torsava Can I close this and archive this repo?

Oh yeah, no problem. Closing.
